### PR TITLE
Add manufacturer data and service data filters explainer

### PR DESCRIPTION
**Pull request by beaufortfrancois**
Please have a look @reillyeon

### CODE REVIEW

File: data-filters-explainer.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+This document covers the use cases identified by web developers asking for
+manufacturer data and service data filtering support and explains the
+specification changes that happened after the [TAG review].
+
+## Use cases
+
+Filtering by device name or prefix may not work for some hardware developers as
+their users rename their devices. In addition, most of their advertised services
+are custom, thus needing 128-bit UUID. Advertising 128-bit UUIDs will occupy
```
Comment (reillyeon) [inline data-filters-explainer.md:11]: "needings 128-bit UUIDs"

File: data-filters-explainer.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+let manufacturerData = [{ companyIdentifier: 0xffff }];
+navigator.bluetooth.requestDevice({ filters: [{ manufacturerData }] });
+```
+
+```js
+let serviceData = [{ service: "battery_service" }];
+navigator.bluetooth.requestDevice({ filters: [{ serviceData }] });
+```
+
+Developers can also provide some data prefix that will filter manufacturer data
```
Comment (reillyeon) [inline data-filters-explainer.md:36]: "provide _a_ data prefix"

File: data-filters-explainer.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+```js
+let manufacturerData = [{
+  companyIdentifier: 0x00e0, /* Google */
+  dataPrefix: new Uint8Array([0x01, 0x02]),
+}];
+navigator.bluetooth.requestDevice({ filters: [{ manufacturerData }] });
+```
+
+A mask can also be used with data prefix to match some patterns in manufacturer
```
Comment (reillyeon) [inline data-filters-explainer.md:49]: "with _a_ data prefix"

**Review by reillyeon — APPROVED** [review]

**Comment by beaufortfrancois** [comment]
@reillyeon Your feedback has been addressed.

File: data-filters-explainer.md (hunk 1 above)
Comment (reillyeon) [reply data-filters-explainer.md]: I made a typo here, it should be "needing 128-bit UUIDs". I added an extra "s".

**Comment by reillyeon** [comment]
Looks good to me (I fixed my own typo). Feel free to merge.

**Pull request merged**